indico/modules/rb/schemas.py

```diff
@@ -401,6 +401,24 @@ def _check_room_name_format_placeholders(self, room_name_format, **kwargs):
             # validated client-side, no i18n needed
             raise ValidationError('Missing placeholders: {}'.format(', '.join(missing)))
 
+        try:
+            placeholders = get_format_placeholders(room_name_format)
+        except ValueError:
+            raise ValidationError(_('Invalid placeholder format'))
+        if invalid := set(placeholders) - {'building', 'floor', 'number'}:
+            # validated client-side, no i18n needed
+            raise ValidationError('Invalid placeholders: {}'.format(', '.join(invalid)))
+
+    @validates('map_url_template')
+    def _check_map_url_template_placeholders(self, map_url_template, **kwargs):
+        try:
+            placeholders = get_format_placeholders(map_url_template)
+        except ValueError:
+            raise ValidationError(_('Invalid placeholder format'))
+        if invalid := set(placeholders) - {'id', 'building', 'floor', 'number', 'lat', 'lng'}:
+            # validated client-side, no i18n needed
+            raise ValidationError('Invalid placeholders: {}'.format(', '.join(invalid)))
```

Comment:
What about adding a `def validate_placeholders(value, placeholders)` util to avoid duplicating all this?